flang/test/Lower/OpenMP/omp-sections.f90

```diff
@@ -0,0 +1,71 @@
+! This test checks the lowering of OpenMP sections construct with several clauses present
+
+! RUN: bbc -fopenmp -emit-fir %s -o - | \
+! RUN: FileCheck %s --check-prefix=FIRDialect
+
+!FIRDialect: func @_QQmain() {
+!FIRDialect: %{{.*}} = fir.alloca i32 {bindc_name = "count", pinned, uniq_name = "_QFEcount"}
+!FIRDialect: %{{.*}} = fir.alloca i32 {bindc_name = "double_count", pinned, uniq_name = "_QFEdouble_count"}
+!FIRDialect: %[[COUNT:.*]] = fir.address_of(@_QFEcount) : !fir.ref<i32>
+!FIRDialect: %[[DOUBLE_COUNT:.*]] = fir.address_of(@_QFEdouble_count) : !fir.ref<i32>
+!FIRDialect: %[[ETA:.*]] = fir.alloca f32 {bindc_name = "eta", uniq_name = "_QFEeta"}
+!FIRDialect: %[[ALLOCATOR:.*]] = arith.constant 1 : i32
+!FIRDialect: omp.sections allocate(%[[ALLOCATOR]] : i32 -> %[[ETA]] : !fir.ref<f32>) {
+!FIRDialect: omp.section {
+!FIRDialect: {{.*}} = arith.constant 5 : i32
+!FIRDialect: fir.store {{.*}} to {{.*}} : !fir.ref<i32>
+!FIRDialect: omp.terminator
+!FIRDialect: }
+!FIRDialect: omp.section {
+!FIRDialect: {{.*}} = fir.load {{.*}} : !fir.ref<i32>
+!FIRDialect: {{.*}} = arith.constant 1 : i32
+!FIRDialect: {{.*}} = arith.addi {{.*}} : i32
+!FIRDialect: fir.store {{.*}} to {{.*}} : !fir.ref<i32>
+!FIRDialect: omp.terminator
+!FIRDialect: }
+!FIRDialect: omp.section {
+!FIRDialect: {{.*}} = fir.load {{.*}} : !fir.ref<f32>
+!FIRDialect: {{.*}} = arith.constant 7.000000e+00 : f32
+!FIRDialect: {{.*}} = arith.subf {{.*}} : f32
+!FIRDialect: fir.store {{.*}} to {{.*}} : !fir.ref<f32>
+!FIRDialect: omp.terminator
+!FIRDialect: }
+!FIRDialect: omp.terminator
+!FIRDialect: }
+!FIRDialect: omp.sections nowait {
+!FIRDialect: omp.terminator
+!FIRDialect: }
+!FIRDialect: return
+!FIRDialect: }
+!FIRDialect: func @_QPfirstprivate(%arg0: !fir.ref<f32>) {
+!FIRDialect: %[[ALPHA:.*]] = fir.alloca f32 {bindc_name = "alpha", pinned, uniq_name = "_QFfirstprivateEalpha"}
+!FIRDialect: %[[ALPHA_STORE:.*]] = fir.load %arg0 : !fir.ref<f32>
+!FIRDialect: fir.store %[[ALPHA_STORE]] to %[[ALPHA]] : !fir.ref<f32> 
+!FIRDialect: omp.sections {
+!FIRDialect: omp.terminator
```

Comment:
Is it OK to have a `sections` operation without a section op? Will it lower to llvm correctly?

Reply:
While lowering to LLVM IR, I have made sure that if there are no section ops within the sections construct, it does not generate anything. So, this should be okay as anyways no LLVM IR will be generated for this code.

Reply:
Can another test be added which has a sections construct with some code, but no section directive? Just to check that a section op will be created for that case.